internal/policy/verify.go

```diff
@@ -201,21 +201,13 @@ func verifyEntry(ctx context.Context, repo *git.Repository, policy *State, entry
 
 		// TODO: evaluate if this can be done once for the earliest commit in
 		// the set being verified if we had them ordered.
-		commitPolicy := policy
-		firstSeenEntry, err := rsl.GetFirstEntryForCommit(repo, commit)
+		var commitPolicy *State
+		commitPolicy, err = GetStateForCommit(ctx, repo, commit)
 		if err != nil {
-			if !errors.Is(err, rsl.ErrNoRecordOfCommit) {
-				return err
-			}
-		} else {
-			commitPolicyEntry, err := rsl.GetLatestEntryForRefBefore(repo, PolicyRef, firstSeenEntry.ID)
-			if err != nil {
-				return err
-			}
-			commitPolicy, err = LoadStateForEntry(ctx, repo, commitPolicyEntry)
-			if err != nil {
-				return err
-			}
+			return err
+		}
+		if commitPolicy == nil {
+			commitPolicy = policy
```

Comment:
This looks like it isn't a change from existing behavior, but this might warrant some more documentation, since it looks like the policy passed in may not be the one that's actually used under certain conditions?

Reply:
Yes, this needs documentation. It's a little related to getting `nil, nil` for when a commit isn't seen before, so the latest policy ought to be used. I'll apply changes to both docs.

Reply:
Done!